src/evaluators/utility/PHAL_SharedParameter.hpp

```diff
@@ -8,27 +8,38 @@
 #include "Phalanx_Evaluator_WithBaseImpl.hpp"
 #include "Phalanx_Evaluator_Derived.hpp"
 #include "Sacado_ParameterAccessor.hpp"
+#include "Albany_Accessors.hpp"
 
 //IKT 6/3/2020 TODO: implement support for vector parameters, which is not available currently.
 
 namespace PHAL
 {
 
-template<typename EvalT, typename Traits, typename ParamNameEnum, ParamNameEnum ParamName>
+template<typename EvalT, typename Traits, typename ParamNameEnum>
 class SharedParameter : public PHX::EvaluatorWithBaseImpl<Traits>,
                         public PHX::EvaluatorDerived<EvalT, Traits>,
                         public Sacado::ParameterAccessor<EvalT, SPL_Traits>
 {
+private:
+  std::shared_ptr<Albany::Accessor<typename EvalT::ScalarT>> accessor;
```

Comment:
If the accessors are created in the problem, this could be a `shared_ptr<const ...>` instead, ensuring SharedParam cannot change the parameter values (you never know with copy+paste in future refactors/changes...).